devel/distcc/Portfile

```diff
@@ -20,14 +42,82 @@ long_description    distcc is a program to distribute compilation of C or \
                     compile but is often 2 to 3 times faster.
 
 homepage            http://distcc.org/
-master_sites        googlecode
-use_bzip2           yes
 
-checksums           rmd160  505447b910d90ea403a98fe0d0a3a0fa92cd7039 \
-                    sha256  311671e844625d7fdb18dd3d096cd855751cfe8de13827682bcb7beff9133b30
+if {${subport} eq "distcc"} {
+    patchfiles-append \
+                    ${PPREFIX}patch-no-fstatat.diff \
+                    ${PPREFIX}patch-no-faccessat.diff \
+                    ${PPREFIX}patch-zeroconf-reg.c.diff \
+                    ${PPREFIX}patch-allow-private-nargs.diff \
+                    ${PPREFIX}patch-whitelistdir.diff
 
-depends_lib         port:popt
+    post-patch {
+        reinplace "s|@PREFIX@|${prefix}|g" \
+                    ${worksrcpath}/src/serve.c \
+                    ${worksrcpath}/src/daemon.c
+        # maintainer convenience: configure.ac was changed so autoreconf
+        # required ... this time (= not every time a `port configure` is run).
+        file delete -force ${worksrcpath}/configure
+    }
+}
 
-configure.args      ac_cv_path_PYTHON= \
+configure.args      --disable-pump-mode \
                     --disable-Werror \
-                    --without-avahi
+                    --enable-rfc2553
+platform darwin {
+    depends_lib     port:popt
+    configure.args-append \
+                    --without-avahi \
+                    --without-libiberty
+}
+
+if {[variant_isset monitor]} {
+    platform darwin {
+        depends_lib-append \
+                    port:gtk2 \
+                    port:libglade2
+    }
+    configure.args-append \
+                    --with-gtk
+    notes-append "Monitoring the distributed build of `port build foo configuredistcc=yes`:\n\
+> env DISTCC_DIR=`port work foo`/.home/.distcc distccmon-gnome"
+}
+if {[variant_isset zeroconf]} {
+    platform darwin {
+        depends_lib-append \
+                    port:avahi
+    }
+    configure.args-replace \
+                    --without-avahi \
+                    --with-avahi
+}
+
+# run autogen.sh if configure isn't present
+use_autoreconf      [expr ![file exists ${worksrcpath}/configure]]
```

Comment:
This will not work. When the Portfile is evaluated, the extract phase has not been run yet. `${worksrcpath}` will never exist and the result of the expression would always be true.